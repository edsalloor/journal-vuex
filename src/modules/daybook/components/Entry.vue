<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: entry.id } })"
  >

    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">{{ day }}</span>
      <span class="mx-1 fs-5">{{ month }}</span>
      <span class="mx-5 fw-light">{{ yearDay }}</span>
    </div>

    <div class="entry-description">
      {{ shortText }}
    </div>
  </div>
</template>

<script>
const months = ['January', 'February', 'March', 'April', 'May', 'June','July', 'August', 'September', 'October', 'November', 'December'];
const days   = ['Sunday','Monday','Tuesday','Wednesday','Thursday','Friday','Saturday'];


export default {
  props: {
    entry: {
      type: Object,
      required: true
    }
  },
  computed: {
    shortText() {
      const entryText = this.entry.text;
      return entryText.length > 130
        ? entryText.substring(0, 130) + '...'
        : entryText;
    },
    day() {
      const date = new Date(this.entry.date);
      return date.getDate();
    },
    month() {
      const date = new Date(this.entry.date);
      return months[date.getMonth()];
    },
    yearDay() {
      const date = new Date(this.entry.date);
      return `${ date.getFullYear() }, ${ days[date.getDay()] }`;
    }
  }
};
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }

  .entry-description {
    font-size: 12px;
  }
}
</style>
