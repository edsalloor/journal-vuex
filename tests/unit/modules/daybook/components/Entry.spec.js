import { shallowMount } from '@vue/test-utils';

import Entry from '@/modules/daybook/components/Entry';

const routerMock = {
  push: jest.fn()
};

const defaultProps = { entry: {
  id: 'DEF456',
  text: 'Hello Friend!',
  date: 1627077227979
}};

const setup = props => shallowMount(Entry, {
  props: { ...defaultProps, props },
  global: {
    mocks: { $router: routerMock }
  }
});

describe('<Entry />', () => {
  const wrapper = setup();

  it('should render', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should redirect on click', () => {
    const entryContainer = wrapper.find('.entry-container');

    entryContainer.trigger('click');

    expect(routerMock.push).toHaveBeenCalledTimes(1);
    expect(routerMock.push).toHaveBeenCalledWith({
      name: 'entry',
      params: { id: defaultProps.entry.id }
    });
  });

  it('computed props', () => {
    expect(wrapper.vm.day).toBe(23);
    expect(wrapper.vm.month).toBe('July');
    expect(wrapper.vm.yearDay).toBe('2021, Friday');
  });
});
