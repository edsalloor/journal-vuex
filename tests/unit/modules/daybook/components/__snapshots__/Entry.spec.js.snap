// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Entry /> should render 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">23</span><span class="mx-1 fs-5">July</span><span class="mx-5 fw-light">2021, Friday</span></div>
  <div class="entry-description">Hello Friend!</div>
</div>
`;
